Dockerfile

```diff
@@ -17,9 +17,35 @@ RUN poetry install --no-dev
 COPY . .
 
 FROM base as prod
-
 ENTRYPOINT poetry run gunicorn --bind 0.0.0.0:5000 "app:create_app(None)"
 
 FROM base as dev
-
 ENTRYPOINT poetry run flask run --host=0.0.0.0
+
+FROM base as test
+RUN poetry install
+
+FROM test as unit
```

Comment:
Purely from a code cleanliness perspective, are these intermediate stages worthwhile?
If you expect the tests to be run separately often this might be valuable - saving the extra build steps if a dev only intends to run the unit tests for example - but we could achieve similar options for the user by always doing the e2e setup, and having an entrypoint like
`poetry run pytest` - the user can then supply arguments when they run the container that add the relevant test directory if they want to run a limited set of tests i.e.
`docker run ... test tests/integration`
If we did go that way, we would want to move our copy of the app/test files though, so that we're not reinstalling chromedriver every time we rebuild to pick up our latest changes.

Food for thought rather than an explicit recommendation

Reply:
Yeah that's a good point- as it happens I've been running the unit tests through docker for dev, so I set it up like this to skip the slow e2e test setup. I guess the file could probably be re-ordered to install the chromedriver first, but that might be awkward to do without including the chromedriver in the prod container, which I'm not keen on doing